homeassistant/components/nextbus/sensor.py

```diff
@@ -0,0 +1,236 @@
+from homeassistant.util.dt import utc_from_timestamp
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'nextbus'
+
+CONF_AGENCY = 'agency'
+CONF_ROUTE = 'route'
+CONF_STOP = 'stop'
+
+ICON = 'mdi:bus'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_AGENCY): cv.string,
+    vol.Required(CONF_ROUTE): cv.string,
+    vol.Required(CONF_STOP): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def validate_value(value_name, value, value_list):
+    """Validate tag value is in the list of items and logs error if not."""
+    valid_values = {
+        v['tag']: v['title']
+        for v in value_list
+    }
+    if value not in valid_values:
+        _LOGGER.error(
+            'Invalid %s tag `%s`. Please use one of the following: %s',
+            value_name,
+            value,
+            ', '.join(
+                '{}: {}'.format(title, tag)
+                for tag, title in valid_values.items()
+            )
+        )
+        return False
+
+    return True
+
+
+def validate_tags(client, agency, route, stop):
+    """Validate provided tags."""
+    # Validate agencies
+    if not validate_value(
+            'agency',
+            agency,
+            client.get_agency_list()['agency'],
+    ):
+        return False
+
+    # Validate the route
+    if not validate_value(
+            'route',
+            route,
+            client.get_route_list(agency)['route'],
+    ):
+        return False
+
+    # Validate the stop
+    route_config = client.get_route_config(route, agency)['route']
+    if not validate_value(
+            'stop',
+            stop,
+            route_config['stop'],
+    ):
+        return False
+
+    return True
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Load values from configuration and initialize the platform."""
+    agency = config[CONF_AGENCY]
+    route = config[CONF_ROUTE]
+    stop = config[CONF_STOP]
+    name = config.get(CONF_NAME)
+
+    from py_nextbus import NextBusClient
+    client = NextBusClient(output_format='json')
+
+    # Ensures that the tags provided are valid, also logs out valid values
+    if not validate_tags(client, agency, route, stop):
+        _LOGGER.error('Invalid config value(s)')
+        return
+
+    add_entities([
+        NextBusDepartureSensor(
+            agency,
+            route,
+            stop,
+            name,
+            client=client,
+        ),
+    ], True)
+
+
+class NextBusDepartureSensor(Entity):
+    """Sensor class that displays upcoming NextBus times.
+
+    To function, this requires knowing the agency tag as well as the tags for
+    both the route and the stop.
+
+    This is possibly a little convoluted to provide as it requires making a
+    request to the service to get these values. Perhaps it can be simplifed in
+    the future using fuzzy logic and matching.
+    """
+
+    def __init__(self, agency, route, stop, name=None, client=None):
```

Comment:
`client` shouldn't be a keyword argument. Make it a positional argument. It will never be None.